tests/UI/Component/Input/InputTest.php

```diff
@@ -410,4 +418,25 @@ public function test_withInput_constraint_fails_and_transformation_different_ord
 		$this->assertEquals($value, $input2->getValue());
 		$this->assertEquals($error, $input2->getError());
 	}
+
+	public function test_withInput_null_transformation_and_constraint() {
+		$values = new DefPostData([]);
+
+		$input = $this->input
+			->withNameFrom($this->name_source)
+			// This currently is the default, but we set it anyway to make out point.
+			->withRequirement(false)
+			->withAdditionalTransformation($this->transformation_factory->custom(function($v) {
+				$this->assertFalse(true); // not called, because there is no input
+			}))
+			->withAdditionalConstraint($this->validation_factory->custom(function($v) {
+				$this->assertFalse(true); // not called, because there is no input
+			}, ""))
+			->withInput($values);
+		$res = $input->getContent();
+
+		$this->assertInstanceOf(Result::class, $res);
+		$this->assertTrue($res->isOk());
+		$this->assertEquals(null, $res->value());
```

Comment:
See [the comment here](https://github.com/klees/ILIAS/pull/37/files#r139443750): This is the exact same problem but from the other side. If there are fields that return null a user won't be able to distinguish a nullary value from a field that deliberately wasn't set.